src/main/java/graphql/util/ParallelTraverser.java

```diff
@@ -0,0 +1,172 @@
+package graphql.util;
+
+import graphql.Internal;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountedCompleter;
+import java.util.concurrent.ForkJoinPool;
+import java.util.function.Function;
+
+import static graphql.Assert.assertNotNull;
+import static graphql.Assert.assertTrue;
+import static graphql.util.TraversalControl.ABORT;
+import static graphql.util.TraversalControl.CONTINUE;
+import static graphql.util.TraversalControl.QUIT;
+
+@Internal
+public class ParallelTraverser<T> {
+
```

Comment:
I knows this is a greg-ism but function<<<<<<>>>> is an annoying type compared to `NodeChildRetriever`